internal/streaming/clickhouse_connector/query.go

```diff
@@ -199,10 +199,10 @@ func (d queryMeterView) toSQL() (string, []interface{}, error) {
 		default:
 			return "", nil, fmt.Errorf("invalid window size type: %s", *d.WindowSize)
 		}
-	}
 
-	if groupByWindowSize {
 		groupByColumns = append(groupByColumns, "windowstart", "windowend")
+	} else {
+		selectColumns = append(selectColumns, "min(windowstart)", "max(windowend)")
```

Comment:
Shouldn't we alias them? I know we reverse-map them by position but we alias other fields

Reply:
Also, do we need to do this in SQL? This `min` and `max` are only used to return the smallest window between the already returned rows. As the returned data is ordered it's basically `rows[0].windowStart` and `rows[len(rows) - 1).windowEnd`, no? Also in many cases when we don't aggregate it's equal with raw windows.

Reply:
Maybe I misunderstood this. This is not the top level windowStart, it's the per value.

Reply:
This piece of code could probably use some refactoring with all the different conditional logic and changes lately.

But the min, max aggregations are only added when there is no grouping by window size which means we get one row per any other group by (or a full total when no grouping is applied).

Reply:
> Shouldn't we alias them? I know we reverse-map them by position in Go but we alias other fields

Unfortunately, not. For that to work, we'd need to give an alias to the table. I'm not sure which one is nicer at the end.

```
SELECT min(windowstart) AS windowstart, max(windowend), sumMerge(value) FROM "om_default_m1"
WHERE windowstart >= '2023-07-31 22:02:00';

Code: 184. DB::Exception: Aggregate function min(windowstart) is found in WHERE in query: While processing min(windowstart) AS windowstart. (ILLEGAL_AGGREGATION) (version 23.8.1.41512 (official build))
```